src/Agent/NewRelic/Agent/Core/Config/configurationPartial.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+    using System.Xml.Serialization;
+    using NewRelic.Agent.Core.Configuration;
+    using NewRelic.Core.Logging;
+
+    namespace NewRelic.Agent.Core.Config;
+
+    // The configuration class is partial.  Part of it is implemented here,
+    // and another part is autogenerated from Configuration.xsd into Configuration.cs.
+    //
+    // Beware the tarpit surrounding the case of property names, as different cases reflect
+    // different origins of the property names.
+    //
+    // Property names such as "agentEnabled" come to us from Configuration.xsd via Configuration.cs.
+    // Property names such as "AgentEnabled" are added in here or inherited from BootstrapConfig.
+    public partial class configuration : IBootstrapConfig
```

Comment:
What if this wasn't a partial class, and this instead took a local configuration in its .ctor. Then the DefaultConfiguration class could take a reference to the Bootstrap configuration. That would make it clearer within the DefaultConfiguration class, that accessing `ServerlessModeEnabled` does more than just grab the local newrelic config value.

Reply:
Maybe? I didn't invent the partial class implementation, just moved it to it's own file. I really think for our purposes at the moment, it's fine as-is. But we could add a tech debt ticket to come back to it at some point.